types/chains.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import "github.com/attestantio/go-eth2-client/spec/phase0"
+
+// DomainType is a unique identifier for signatures, 2 identical pieces of data signed with different domains will result in different sigs
+type DomainType [4]byte
+
+// Chains are intended to separate different SSV chains. A chain can have many forks in it.
+const (
+	MainnetChain = 0x0
+	PrimusChain  = 0x1
+	ShifuChain   = 0x2
+	JatoChain    = 0x3
+)
+
+// DomainTypes represent specific forks for specific chains, messages are signed with the domain type making 2 messages from different domains incompatible
+var (
+	GenesisMainnet = DomainType{0x0, 0x0, MainnetChain, 0x0}
+	PrimusTestnet  = DomainType{0x0, 0x0, PrimusChain, 0x0}
+	ShifuTestnet   = DomainType{0x0, 0x0, ShifuChain, 0x0}
+	ShifuV2Testnet = DomainType{0x0, 0x0, ShifuChain, 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, JatoChain, 0x1}
+)
+
+// ForkData is a simple structure holding fork information for a specific chain (and its fork)
+type ForkData struct {
+	// Epoch in which the fork happened
+	Epoch phase0.Epoch
+	// Domain for the new fork
+	Domain DomainType
+}
+
+func (domainType DomainType) GetChain() byte {
+	return domainType[2]
+}
+
+func (domainType DomainType) GetForksData() []*ForkData {
```

Comment:
where is this used?

Reply:
It should from Runner according to the SIP: https://github.com/bloxapp/SIPs/pull/26/files#diff-ef3380c5833665d8dd7ac5bc173946735e412f9da5ca4c0380d39399b5458af6L75

but it isn't part of this PR for some reason...
@alonmuroch is there a reason for this?

Reply:
not used yet